### DISCUSSION TITLE
alte Version wird angezeigt

### DISCUSSION THREAD

**Discussion by woupi** [camueller/SmartApplianceEnabler#478] [Installation und Betrieb]
Ich habe gerade das update von 2.1.0 auf 2.2.1 gemacht.
Also 
- Datei "SmartApplianceEnabler-2.2.1.war" runtergeladen
- alte Datei, im Ordner /opt/sae/  umbenannt in "SmartApplianceEnabler-2.1.0.war.alt (WinSCP mit sudo Rechten)
- neue Datei in Ordner /opt/sae/ hochgeladen (WinSCP mit sudo Rechten)
- Device2EM.xml und Appliances.xml gesichert (WinSCP mit sudo Rechten)
- sudo systemctl restart smartapplianceenabler.service
- SAE im Browser neu aufgerufen (Versionsnummer 2.1.0)
- Cache gelöscht  (Versionsnummer 2.1.0)
- lxc SAe in Proxmox neu gestartet
- SAE im Browser neu aufgerufen (Versionsnummer 2.1.0)
- Cache gelöscht  (Versionsnummer 2.1.0)
- alte Datei runter geladen
- SAE im Browser neu aufgerufen (Versionsnummer 2.1.0)
- Cache gelöscht  (Versionsnummer 2.1.0)
Es bleibt dabei, es wird die alte Version angezeigt und der Hinweis auf die neue Version ist natürlich auch da.
Axel, was kann ich falsch gemacht haben? Mir fällt nichts ein.
Danke schonmal im Voraus.
VG
Sascha

**Comment by camueller** [comment]
Ich kenne Proxmox nicht. Wenn da Docker-Container verwendet werden, musst Du das neue Docker-Image verwenden.

Mein Standpunkt ist, dass man Container nur verwenden sollte, wenn man sich damit auskennt.

**Comment by woupi** [comment]
Sind keine Docker-Container. Docker-Container habe ich schnell an acta gelegt, da ich mich damit einfach Zuviel hätte beschäftigen müssen.

> Mein Standpunkt ist, dass man Container nur verwenden sollte, wenn man sich damit auskennt.

Im Groh habe ich mich mit Proxmox, VM und lxc beschäftigt. Ich bin sicher weit weg von einem Profi und würde auch nie behaupten das ich mich damit auskenne, aber das geht jedem so wenn er anfängt, ansonsten hätte ich weder influxdb, grafana, Linux, sae usw. 
Aber der lxc benutzt, wie bei containern üblich, große Teile des Wirtsystems und kann so abgespeckt installiert werden. Proxmox basiert auf Debian. lxc ist halt schlanker als VM. Auf dem lxc läuft Ubuntu 22.04.2 LTS Server, also eigentlich nichts besonderes.

Bis jetzt hat auch jedes update einwandfrei funktioniert, auch das große von 1.x auf 2.x.

Wo es jetzt hakt, keine Ahnung.

**Comment by woupi** [comment]
Ich muss leider nochmal zurück kommen, da ich es immer noch nicht hin bekommen habe. 
Ich habe jetzt die Version 2.2.2 hochgeladen.
Anbei ein Bild von dem Ordner, eventuell erkennt ja daran einer, etwas was falsch ist.
Mir ist bewusst, dass dies keine Raspberry Installation ist, aber bisher haben alle Updates immer funktioniert, also bis auf die ab 2.2.1.
![image](https://user-images.githubusercontent.com/98451301/234178141-cf104603-4a87-424e-a450-cd317f6ba5d3.png)

Mir ist auch bewusst, dass das kein Fehler von SAE ist, da es ja offensichtlich bei allen anderen funktioniert.
Trotzdem würde ich natürlich gerne herausfinden woran das liegt und bin dabei auf eure Hilfe, vor allem natürlich von Axel, keiner kennt SAE so gut wie der Programmierer selbst, aber auch von anderen, angewiesen.

Ich bedanke mich im Voraus für alle Tipps.
VG

**Comment by andi5783** [comment]
Hi, ich kenne dein Setup auch nicht, aber evtl. hilft das:
Beim Einrichten von Docker werden auch Environment-Variablen gesetzt, z.B. 
`SAE_VERSION 2.2.2` 

d.h. bei mir reicht auch nicht das bloße kopieren der neuen Version, sondern ich muss die Variable auch anpassen. Evtl. ist das Problem bei dir etwas ähnliches.

**Comment by woupi** [comment]
Vielen Dank, für den Versuch zu Helfen.
Möglich wäre das, allerdings habe ich mit der Konstellation (Proxmox, lxc, SAE) schon updates gefahren (z.B. von 1.9 auf 2.0) und dabei gab es nicht das Problem.
Darum vermute ich, dass es das nicht sein kann, habe aber mal im Proxmox Forum nachgefragt.
Gefunden habe ich passendes, weder im Proxmox Forum noch bei google.
Ich werde mal abwarten, ob sich da eventuell einer meldet.
Vielen Dank nochmal.
VG
